augur/dates/ambiguous_date.py

```diff
@@ -131,3 +149,31 @@ def assert_only_less_significant_uncertainty(self):
                 raise InvalidDate(self.uncertain_date,
                     "Month contains uncertainty, so day must also be uncertain."
                 )
+
+
+def get_bounds(min_max_year):
+    """Get exact date bounds based on given years."""
+    # This must be an iterable with at least one value.
+    assert len(min_max_year) > 0
+
+    if len(min_max_year) > 2:
+        raise AugurError(f"The year bounds {min_max_year!r} must have only one (lower) or two (lower, upper) bounds.")
+
+    try:
+        lower_bound = datetime.date(int(min_max_year[0]), 1, 1)
+    except:
+        raise AugurError(f"{min_max_year[0]!r} is not a valid year.")
+
+    if len(min_max_year) == 2:
+        try:
+            upper_bound = datetime.date(int(min_max_year[1]), 12, 31)
+        except:
+            raise AugurError(f"{min_max_year[1]!r} is not a valid year.")
```

Comment:
_non-blocking_

I have a general question about raising `AugurError` within `augur/dates`. 

Should these be internal errors that are then handled by the augur subcommands that call on the dates module? This gives room for each subcommand to be more specific about which exact CLI option needs to be changed correct the error.

Reply:
That sounds like a good idea. Currently `InvalidDate` is the "internal error" but it still gets handled within the dates module. Instead of handling it within the dates module, it could be handled in each subcommand with more specific error messages like you mentioned.

The errors here are a bit different since it's not about the date, but the argument `min_max_year`. With the approach above, it might make sense to create a new exception class (e.g. `InvalidYearBounds`) which is handled wherever this argument is used.

Reply:
A new exception class sounds like a good plan!

Reply:
Updated to use `InvalidYearBounds`.